hardware/esp8266com/esp8266/libraries/ESP8266WiFi/examples/NTPClient/NTPClient.ino

```diff
@@ -67,20 +67,39 @@ void setup()
   Serial.println(udp.localPort());
 }
 
+
+
 void loop()
 {
+  // this will keep track of answers we get. If we don't get answers, one reasion might be that the udp receiver 
+  // might not be working any more (for example due to interrupted and restored WIFI connection)
+  static int udperrorcounter = 0;
+
   //get a random server from the pool
-  WiFi.hostByName(ntpServerName, timeServerIP); 
+  if (!WiFi.hostByName(ntpServerName, timeServerIP)) {
+    Serial.println("Hostname lookup failed");
+    return;
+  }
 
   sendNTPpacket(timeServerIP); // send an NTP packet to a time server
   // wait to see if a reply is available
+  udperrorcounter++;
   delay(1000);
-  
+
   int cb = udp.parsePacket();
   if (!cb) {
     Serial.println("no packet yet");
+
+    if (udperrorcounter > 3) { // 3 is just randomly choosen. in case of lot packet drops...
```

Comment:
based on the error it can be the case that only the UDP socket for NTP is closed, but other sockets can still be alive. so with ```udp.stopAll()``` you will kill them all.

the bool compare is overloaded for the WiFiUdp class:
https://github.com/esp8266/Arduino/blob/6422688cc61115b20cdd7fb32be97cb296f977e2/hardware/esp8266com/esp8266/libraries/ESP8266WiFi/src/WiFiUdp.h#L42
and it will check if exactly this socket it alive or not.